test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -50,24 +50,77 @@ struct value_t {
 /* A collection is made of mapped Key objects. */
 struct collection_t {
     std::map<key_value_t, key_t> keys;
-    std::map<key_value_t, value_t> *values = {nullptr};
+    std::map<key_value_t, value_t> values;
 };
 
 /* Representation of the collections in memory. */
 class database {
     public:
-    const std::vector<std::string>
-    get_collection_names() const
+    std::vector<std::string>
+    get_collection_names()
     {
+        std::lock_guard<std::mutex> lg(_mtx);
         std::vector<std::string> collection_names;
 
-        for (auto const &it : collections)
+        for (auto const &it : _collections)
             collection_names.push_back(it.first);
 
         return (collection_names);
     }
 
-    std::map<std::string, collection_t> collections;
+    std::map<key_value_t, key_t>
+    get_keys(const std::string &collection_name)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return (_collections.at(collection_name).keys);
+    }
+
+    void
+    add_collection(const std::string &collection_name)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        testutil_assert(_collections.find(collection_name) == _collections.end());
```

Comment:
Should we have a similar assert in the other methods ? (get_record -> check the collection and the key exist ?)

Reply:
It isn't necessary because we're using `at` which will throw if the key doesn't exist.